src/main/java/by/issoft/training/authorization/TokenGenerator.java

```
@@ -0,0 +1,76 @@
+    private TokenGenerator() {
+    }
+
+    //Method to create new token if it does not exist
+    public static TokenGenerator getInstance() {
+        if (instance == null) {
+            instance = new TokenGenerator();
+        }
+        return instance;
+    }
+
+    public HashMap<Scope, String> createAccessToken(Scope scope) {
+        try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+            FileInputStream inputStream = new FileInputStream("src/main/resources/config.properties");
+            Properties properties = new Properties();
+            properties.load(inputStream);
+            HttpPost request = new HttpPost(properties.getProperty("url") + "/oauth/token");
+            String auth = properties.getProperty("user_name") + ":" + properties.getProperty("password");
+            byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+            String authHeader = "Basic " + new String(encodedAuth);
+            request.setHeader(HttpHeaders.AUTHORIZATION, authHeader);
+            request.setHeader(HttpHeaders.CONTENT_TYPE, "application/x-www-form-urlencoded");
+            List<NameValuePair> nameValuePair = new ArrayList<>();
+            nameValuePair.add(new BasicNameValuePair("grant_type", properties.getProperty("grant_type")));
+            nameValuePair.add(new BasicNameValuePair("scope", scope.getScopeStringValue()));
+            request.setEntity(new UrlEncodedFormEntity(nameValuePair));
+            ResponseHandler<String> responseHandler = new HandlerOfResponse();
+            String httpResponse = httpclient.execute(request, responseHandler);
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode node = mapper.readTree(httpResponse);
+            String token = node.get("access_token").asText();
+            tokensStorage.put(scope, token);
+
+        } catch (IOException ex) {
+            ex.printStackTrace();
+            System.exit(1);
```

Throw RuntimeException with readable message

---

Added throwing RuntimeException instead of printing stack trace and exit.